### ISSUE TITLE
As an organisation admin, I want to be able to allow all my organisation's domain users to access one on my organisation's applications

### ISSUE THREAD

**Issue by FlorisFokkinga** [SURFscz/SBS#1510]
### Use case
An institution wants to give all their users access to an application, and wants to enable their users to invite users from outside their institution to collaborate with them.
This is very similar to something SURFconext + Invite already offers, but with a focus on self service and collaboration.

### Design
This will only work for applications that are owned by one of the organisations on the platform, currently only editable by platform admins.

<img width="1223" alt="github_1500" src="https://github.com/user-attachments/assets/a2dea572-daa8-4464-a7af-443f5f035121">

**Comment by logan-life** [comment]
Q: How to handle domains for consortiums?

**Comment by logan-life** [comment]
For organizations, the solution is straightforward: a single domain is associated with the org, and users that login with that domain are then allowed to access the org application.

**Comment by logan-life** [comment]
For consortia, a potential solution would be that a platform admin can edit the consortium's organization domains and add all the domains of the consortium members. 

Question is, would this have any unwanted consequences on other functionality.

**Comment by logan-life** [comment]
The consortia would need to get approval from the member institutions before a platform admin would add the member institution domains to the organization configuration. Responsibility for getting that permission should stay with the consortium.

**Comment by oharsta** [comment]
And with regards to the consortia use-case: currently a domain name must be unique and can not be shared among organisations. This constraint has to be removed in order to support consortia?

**Comment by FlorisFokkinga** [comment]
1. Not yet, I'd say it's OK to implement it simple first. @logan-life?
2. Only for platform admins, unchanged.
3. Correct.
4. Yes, but only if their service is assigned to an organisation, which only platform admins can do.
5. No one should be able to access the service.
6. Let's keep that out of scope of this issue.

**Comment by logan-life** [comment]
Regarding #6, that's something we do need to consider side effects I think. Maybe there was some functional or other reason why the unique-ness constraint for DNs was applied initially. Removing constraints from a DB entry is something to touch carefully in my experience 

For the rest, agreed with Floris

**Comment by oharsta** [comment]
Let me rephrase question 1: is there any situation where we would link a service to an organisation and not grant domain users access? That is I thought the whole point of linking a service to an organisation. Then why bother with the option to disable the access for domain users explicit if a platform admin can just disable the link. If we want to grant a service admin the ability to not grant domain users access, then why link the service in the first place? E.g. do we need to make UI changes?

**Comment by FlorisFokkinga** [comment]
I think we should make the proposed UI change, because it will show the application admin the current state. It also gives the application admin some control.

**Comment by mrvanes** [comment]
Seems to work fine, but how is someone that has access by org via domain going to invite "users from outside their institution to collaborate with them"? There is no CO to invite anyone to, let alone they are admin to do so?

**Comment by mrvanes** [comment]
Discussed during personeelsdag, invitation is then done via newly created CO.
Done!
